backend/lucene/src/main/java/org/hibernate/search/backend/lucene/search/extraction/impl/ExtractionRequirements.java

```diff
@@ -63,19 +71,20 @@ public LuceneCollectors createCollectors(IndexSearcher indexSearcher, Query orig
 			totalHitCountThreshold = requestedTotalHitCountThreshold;
 		}
 
-		TopDocsCollector<?> topDocsCollector = null;
+		CollectorManager<? extends TopDocsCollector<?>, ? extends TopDocs> topDocsCollectorManager = null;
 		Integer scoreSortFieldIndexForRescoring = null;
 		boolean requireFieldDocRescoring = false;
 
 		CollectorExecutionContext executionContext =
 				new CollectorExecutionContext( metadataResolver, indexSearcher, maxDocs );
 
-		CollectorSet.Builder collectorsForAllMatchingDocsBuilder =
-				new CollectorSet.Builder( executionContext, timeoutManager );
+		HibernateSearchMultiCollectorManager.Builder collectorsForAllMatchingDocsBuilder =
+				new HibernateSearchMultiCollectorManager.Builder( executionContext, timeoutManager );
 
 		if ( maxDocs > 0 ) {
 			if ( sort == null || isDescendingScoreSort( sort ) ) {
-				topDocsCollector = TopScoreDocCollector.create( maxDocs, totalHitCountThreshold );
+				topDocsCollectorManager = new HibernateSearchTopScoreDocCollectorManager( offset, maxDocs, null,
```

Comment:
note this `null` .... 

> after – the previous doc after which matching docs will be collected.

I guess we could try to leverage that, but it would only work in the case of scrolling since,  as with a simple offset, we don't have the doc...

Reply:
We can't leverage that right now, it's a separate feature that would need its own API, and even then it won't change anything for users that don't explicitly opt-in that feature.

See https://hibernate.atlassian.net/browse/HSEARCH-2601

Reply:
Also we can't use that for scrolling in all conditions since it requires unique sort keys.

Let alone the performance implications of going from an offset to a "searchAfter". I'm pretty sure the performance profile would be quite different.